cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/DeletedProgramHandlerStage.java

```diff
@@ -91,35 +89,13 @@ public void process(ApplicationDeployable appSpec) throws Exception {
       ProgramType type = ProgramTypes.fromSpecification(spec);
       final ProgramId programId = appSpec.getApplicationId().program(type, spec.getName());
       programTerminator.stop(programId);
+      programScheduler.deleteSchedules(programId);
+      programScheduler.modifySchedulesTriggeredByDeletedProgram(programId);
```

Comment:
what happens when this happens at the same time that some other app is deleted, which then tries to modify the same schedules?